lib/baby_squeel/compat.rb

```diff
@@ -163,5 +170,24 @@ def where(*args, &block)
         end
       end
     end
+
+    module Resolver
+      private
+
+      def build(strategy, name, *args)
+        result = super(strategy, name, *args)
+        result.nil? && strategy == :table ? Table.new(Table.new(Arel::Table.new(name))) : result
+      end
+
+      def valid_name?(strategy, name)
+        result = super(strategy, name)
+        result.nil? && strategy == :table ? true : result
```

Comment:
Because the super implementation doesn't handle `:table`, you could probably just do this:

```ruby
strategy == :table || super(strategy, name)
```